dotnet/test/common/RelativeLocatorTest.cs

```diff
@@ -53,14 +75,166 @@ public void ShouldBeAbleToCombineFilters()
+                                + "    \theight: 50px;\n"
+                                + "    \twidth: 50px;\n"
+                                + "    }"
+                )
+               .WithBody(
+                  "<span style=\"position: relative;\">\n"
+                                + "    <div id= \"a\" class=\"c\" style=\"left:25;top:0;\">El-A</div>\n"
+                                + "    <div id= \"b\" class=\"c\" style=\"left:78;top:30;\">El-B</div>\n"
+                                + "    <div id= \"c\" class=\"c\" style=\"left:131;top:60;\">El-C</div>\n"
+                                + "    <div id= \"d\" class=\"c\" style=\"left:0;top:53;\">El-D</div>\n"
+                                + "    <div id= \"e\" class=\"c\" style=\"left:53;top:83;\">El-E</div>\n"
+                                + "    <div id= \"f\" class=\"c\" style=\"left:106;top:113;\">El-F</div>\n"
+                                + "  </span>"
+                ));
+
+            IWebElement @base = driver.FindElement(By.Id("e"));
+            ReadOnlyCollection<IWebElement> cells = driver.FindElements(RelativeBy.WithLocator(By.TagName("div")).Above(@base));
+
+            IWebElement a = driver.FindElement(By.Id("a"));
+            IWebElement b = driver.FindElement(By.Id("b"));
+
+            var values = cells.Select(element => element.GetDomAttribute("id"));
+            Assert.That(values, Is.EqualTo(new List<string> { b.GetDomAttribute("id"), a.GetDomAttribute("id") }));
+        }
+
+        [Test]
+        public void NearLocatorShouldFindNearElements()
+        {
+            driver.Url = (EnvironmentManager.Instance.UrlBuilder.WhereIs("relative_locators.html"));
+
+            var rect1 = driver.FindElement(By.Id("rect1"));
+
+            var rect2 = driver.FindElement(RelativeBy.WithLocator(By.Id("rect2")).Near(rect1));
+
+            Assert.That(rect2.GetDomAttribute("id"), Is.EqualTo("rect2"));
+        }
+
+        [Test]
+        public void NearLocatorShouldNotFindFarElements()
+        {
+            driver.Url = (EnvironmentManager.Instance.UrlBuilder.WhereIs("relative_locators.html"));
+
+            var rect3 = driver.FindElement(By.Id("rect3"));
+
+            Assert.That(() =>
             {
-                string id = element.GetAttribute("id");
-                elementIds.Add(id);
-            }
+                var rect2 = driver.FindElement(RelativeBy.WithLocator(By.Id("rect4")).Near(rect3));
 
-            Assert.That(elementIds, Is.EquivalentTo(new List<string>() { "third" }));
+            }, Throws.TypeOf<NoSuchElementException>().With.Message.EqualTo("Unable to find element; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception"));
+        }
+
+        //------------------------------------------------------------------
+        // Tests below here are not included in the Java test suite
```

Comment:
Do not mention Java in source code, we rely on it to increase coverage, but it is not one-to-one relationship. Thx.

Reply:
I was copying the existing style, should all those be removed?

![image](https://github.com/user-attachments/assets/d216886c-feef-4a5d-ac1d-c188618cee77)

Reply:
LOL, no comments :D I guess there is no search entry for Ruby or Python. Java is the best :)

Seriously, keep it as it, separate topic.